### PR TITLE
[HAL-01] Better attached value handling.

### PR DESCRIPTION
**Pull request by zie1ony**
This changes the behavoior of handling attached value.
Only payable entry points checks the attached value.

### CODE REVIEW

File: odra-casper/backend/src/utils.rs [hunk 1]
```diff
@@ -49,26 +49,15 @@ pub fn clear_attached_value() {
...
-
-/// Reverts with an [ExecutionError] if some value is attached to the call.
-pub fn assert_no_attached_value() {
-    if named_arg_exists(consts::CARGO_PURSE_ARG) {
-        let cargo_purse = runtime::get_named_arg(consts::CARGO_PURSE_ARG);
-        let amount = casper_contract::contract_api::system::get_purse_balance(cargo_purse);
-        if amount.is_some() && !amount.unwrap().is_zero() {
-            revert(ExecutionError::non_payable());
-        }
+    let cargo_purse = runtime::get_named_arg(consts::CARGO_PURSE_ARG);
```
Comment (kpob) [inline odra-casper/backend/src/utils.rs:52]: What happens if the `cargo purse` arg is not attached?

Comment (zie1ony) [reply odra-casper/backend/src/utils.rs:52]: It panics with standard Casper missing argument way. Exactly the same as it pancis with missing an argument.

Comment (kpob) [reply odra-casper/backend/src/utils.rs:52]: Hmm, strange a bit, you think, you've passed all the required args, according to the signature, but you get the MissingArg err. But let's keep it.

**Review by kpob — APPROVED** [review]

**Comment by kpob** [comment]
@zie1ony Seems it can't be like that. Now a payable function revets if no value is attached.

**Comment by kpob** [comment]
Btw. reverting with the `MissingArgument` error wasn't a good choice. It took about 2h to find out what caused the error., I checked a few times all the args required by the entry point, then a few different places in the code. Missing a cargo purse was the last resort.

**Pull request merged**